compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2362,6 +2376,26 @@ fn add_upstream_rust_crates<'a, B: ArchiveBuilder<'a>>(
                             (lib.name, lib.kind, lib.verbatim)
                         };
 
+                        if sess.opts.unstable_opts.split_bundled_libs {
+                            if let NativeLibKind::Static {
+                                bundle: Some(true) | None,
+                                whole_archive,
+                            } = lib.kind
+                            {
+                                let suffix = &sess.target.staticlib_suffix;
+                                let crate_path = src.paths().next().unwrap().to_str().unwrap();
```

Comment:
```suggestion
                                let crate_path = src.rlib.as_ref().unwrap().0;
```
`fix_windows_verbatim_for_gcc` may also be necessary.
See how `link_rlib` is used in `fn add_static_crate`.